Realm/Realm/Exceptions/ErrorCode.cs

```diff
@@ -18,6 +18,16 @@
 
 namespace Realms.Sync.Exceptions
 {
+
+    // ATTENTION: given that managed does not differentiate errors by category like core does,
```

Comment:
Might be worth confirming this with someone in Core, but as far as I remember, client error codes are in the 100-199 range, while protocol errors are 200-299, so you can use the value to infer the category.

Reply:
I could see that [Protocol Error](https://github.com/realm/realm-core/blob/9874d5164927ea39273b241a5af14b596a3233e9/src/realm/sync/protocol.hpp#L197) has `connection level errors` and `session level errors`. `connection level errors` use the 100-199, so that'd collide with the [ClientError](https://github.com/realm/realm-core/blob/9874d5164927ea39273b241a5af14b596a3233e9/src/realm/sync/client_base.hpp#L73) which also use the 100-199 range. Because of all of this I decided to go with the solution you see here.
In case I misunderstood something I'll ask core about it.
Or if you think we'll never want to pass `connection level errors` then, yes we can avoid all this. But "never want to" seems unlikely to me.